mmv1/third_party/terraform/services/compute/resource_compute_future_reservation_test.go.tmpl

```diff
@@ -0,0 +1,99 @@
+{{- if ne $.TargetVersionName "ga" -}}
+package compute_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+)
+
+func TestAccComputeFutureReservation_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+		"project":       envvar.GetTestProjectFromEnv(),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderBetaFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeFutureReservation_full(context),
+			},
+			{
+				ResourceName:            "google_compute_future_reservation.gce_future_reservation",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"auto_delete_auto_created_reservations"},
+			},
+			{
+				Config: testAccComputeFutureReservation_update(context),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("google_compute_future_reservation.gce_future_reservation", plancheck.ResourceActionUpdate),
+					},
+				},
+			},
+			{
+				ResourceName:            "google_compute_future_reservation.gce_future_reservation",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"auto_delete_auto_created_reservations"},
+			},
+		},
+	})
+}
+
+func testAccComputeFutureReservation_full(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_compute_future_reservation" "gce_future_reservation" {
+  provider = google-beta
+  name     = "tf-fr%{random_suffix}"
+  name_prefix = "fr-%{random_suffix}"
+  project  = "%{project}"
+  auto_delete_auto_created_reservations = true
+  description      = "test future reservation"
+  time_window {
+    start_time = "2025-08-01T00:00:00Z"
+    end_time   = "2025-08-02T00:00:00Z"
+  }
```

Comment:
It looks like there are a lot of fields not marked as immutable that could be added here. For example, `planning_status` is defined as a field that can be specified by the user, and `reservation_name` is defined to be mutable.